stdlib/LibGit2/src/diff.jl

```diff
@@ -109,23 +109,25 @@ are to be included or not).
 """
 function deletions(diff_stat::GitDiffStats)
     ensure_initialized()
-    return ccall((:git_diff_stats_deletions, libgit2), Csize_t, (Ptr{Cvoid},), diff_stat.ptr)
+    return ccall((:git_diff_stats_deletions, libgit2), Csize_t, (Ptr{Cvoid},), diff_stat)
 end
 
 function count(diff::GitDiff)
     ensure_initialized()
-    return ccall((:git_diff_num_deltas, libgit2), Cint, (Ptr{Cvoid},), diff.ptr)
+    return ccall((:git_diff_num_deltas, libgit2), Cint, (Ptr{Cvoid},), diff)
 end
 
 function Base.getindex(diff::GitDiff, i::Integer)
     if i < 1 || i > count(diff)
         throw(BoundsError(diff, (i,)))
     end
     ensure_initialized()
-    delta_ptr = ccall((:git_diff_get_delta, libgit2),
-                      Ptr{DiffDelta},
-                      (Ptr{Cvoid}, Csize_t), diff.ptr, i-1)
-    return unsafe_load(delta_ptr)
+    GC.@preserve diff begin
```

Comment:
Shoud use unsafe_convert

Reply:
I can replace `diff.ptr` by `diff` but I think that `preserve` is needed because of the `unsafe_load` to avoid that `delta_ptr` becomes invalid if `diff` object is garbage collected (unless it is safe to assume that `delta_ptr` may still be ok).

Reply:
Good catch. Remove the `.ptr`, keep the `preserve`, but add a comment to explain that `delta_ptr` inherits ownership from `diff`, so `diff` needs to be preserved.